chain/client/src/rocksdb_metrics.rs

```diff
@@ -0,0 +1,151 @@
+        Ok(_) => {}
+        Err(err) => {
+            warn!(target:"stats", "Failed to export store statistics: {:?}",err);
+        }
+    }
+}
+
+/// Wrapper for re-exporting RocksDB stats into Prometheus metrics.
+pub(crate) static ROCKSDB_METRICS: Lazy<Mutex<RocksDBMetrics>> =
+    Lazy::new(|| Mutex::new(RocksDBMetrics::default()));
+
+#[derive(Default, Debug)]
+/// Creates prometheus metrics on-demand for exporting RocksDB statistics.
+pub(crate) struct RocksDBMetrics {
+    // Contains counters and sums, which are integer statistics in RocksDB.
+    int_gauges: HashMap<String, IntGauge>,
+    // Contains floating point statistics, such as quantiles of timings.
+    gauges: HashMap<String, GaugeVec>,
+}
+
+impl RocksDBMetrics {
+    pub fn export_stats_as_metrics(
+        &mut self,
+        stats: StoreStatistics,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        for (stat_name, values) in stats.data {
+            if values.len() == 1 {
+                // A counter stats.
+                // A statistic 'a.b.c' creates the following prometheus metric:
+                // - near_a_b_c
+                if let StatsValue::Count(value) = values[0] {
+                    self.set_int_value(
+                        |stat_name: &str| stat_name.to_string(),
+                        |stat_name| get_prometheus_metric_name(stat_name),
+                        &stat_name,
+                        value,
+                    )?;
+                }
+            } else {
+                // A summary stats.
+                // A statistic 'a.b.c' creates the following prometheus metrics:
+                // - near_a_b_c_sum
+                // - near_a_b_c_count
+                // - near_a_b_c{quantile="0.95"}
+                for stats_value in values {
+                    match stats_value {
+                        StatsValue::Count(value) => {
+                            self.set_int_value(
+                                |stat_name| get_stats_summary_count_key(stat_name),
+                                |stat_name| get_metric_name_summary_count_gauge(stat_name),
+                                &stat_name,
+                                value,
+                            )?;
+                        }
+                        StatsValue::Sum(value) => {
+                            self.set_int_value(
+                                |stat_name| get_stats_summary_sum_key(stat_name),
+                                |stat_name| get_metric_name_summary_sum_gauge(stat_name),
+                                &stat_name,
+                                value,
+                            )?;
+                        }
+                        StatsValue::Percentile(percentile, value) => {
+                            let key = &stat_name;
+
+                            let entry = self.gauges.entry(key.to_string());
+                            match entry {
```

Comment:
It seems this can be compressed:

```
let gauge = match entry {
  Vacant(entry) => entry.insert(create()?),
  Occupied(entry) => entry.get_mut(),
};
gauge.with_label_values…
```